### ISSUE TITLE
@nrwl/next needed as dependency rather than devDependency

### ISSUE THREAD

**Issue by ckaeslin** [nrwl/nx#16575]
### Current Behavior

Probably a regression of https://github.com/nrwl/nx/issues/12150

@nrwl/next in package.json dependency leads to errors when running in production with npm install --omit=dev  --prefer-offline --no-audit --ignore-scripts

### Expected Behavior

@nrwl/next can be put into devDependencies and the app runs

### Steps to Reproduce

1. Create a simple new nx next app.
2. Dockerize it with the following docker file.

```
FROM node:18-alpine
ENV NODE_ENV=production
WORKDIR /app

COPY package.json package-lock.json ./
RUN npm install --omit=dev  --prefer-offline --no-audit --ignore-scripts
```

### Nx Report

```shell
Node : 18.16.0
OS   : darwin arm64
npm  : 9.6.5

nx                      : 15.9.2
@nrwl/js                : 15.9.2
@nrwl/jest              : 15.9.2
@nrwl/linter            : 15.9.2
@nrwl/workspace         : 15.9.2
@nrwl/cli               : 15.9.2
@nrwl/cypress           : 15.9.2
@nrwl/devkit            : 15.9.2
@nrwl/eslint-plugin-nx  : 15.9.2
@nrwl/nest              : 15.9.2
@nrwl/next              : 15.9.2
@nrwl/node              : 15.9.2
@nrwl/nx-plugin         : 15.9.2
@nrwl/react             : 15.9.2
@nrwl/rollup            : 15.9.2
@nrwl/storybook         : 15.9.2
@nrwl/tao               : 15.9.2
@nrwl/web               : 15.9.2
@nrwl/webpack           : 15.9.2
typescript              : 5.0.4
---------------------------------------
Community plugins:
@nx-extend/strapi : 3.5.1
@nxkit/playwright : 2.2.0
---------------------------------------
Local workspace plugins:
```

### Failure Logs

```shell
error - Failed to load next.config.js, see more info here https://nextjs.org/docs/messages/next-config-error
Error: Cannot find module '@nrwl/next/plugins/with-nx'
Require stack:
- /app/next.config.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at mod._resolveFilename (/app/node_modules/next/dist/build/webpack/require-hook.js:23:32)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (/app/next.config.js:2:16)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/app/next.config.js' ]
}
```

**Comment by ckaeslin** [comment]
just saw it will probably be fixed by https://github.com/nrwl/nx/pull/16469

**Comment by AgentEnder** [comment]
Yup, this is already fixed by #16469

**Comment by wingy3181** [comment]
Is this fix only available in 16?
